src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -110,7 +115,9 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
-        outState.putSerializable("device", mDevice);
+        if (mIsCreate && mDevice != null) {
```

Comment:
This isn't needed (you can store null, and who cares if we save/restore a device and later replace it in `#onApiChange`?)